### PR TITLE
add e2e test for volume resizing

### LINKED ISSUE #316: Add e2e test for volume resize
**Is your feature request related to a problem?/Why is this needed**
Add e2e test for volume resize

### PR DESCRIPTION
**Pull request by AndyXiangLi**
**Is this a bug fix or adding new feature?**
Fixes #316
**What is this PR about? / Why do we need it?**
Add e2e test for volume resizing

### CODE REVIEW

File: tests/e2e/testsuites/dynamically_provisioned_resize_volume_tester.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+	By("resizing the pvc")
+	updatedPvc, err := client.CoreV1().PersistentVolumeClaims(namespace.Name).Update(pvc)
+	if err != nil {
+		framework.ExpectNoError(err, fmt.Sprintf("fail to resize pvc(%s): %v", pvcName, err))
+	}
+	updatedSize := updatedPvc.Spec.Resources.Requests["storage"]
+
+	By("Wait for resize to complete")
+	time.Sleep(30 * time.Second)
```
Comment (ayberk) [inline tests/e2e/testsuites/dynamically_provisioned_resize_volume_tester.go:63]: How did you come up with this number? Can we do this incrementally? Check every 5 seconds and give up after 1 minute for example.

Comment (AndyXiangLi) [reply tests/e2e/testsuites/dynamically_provisioned_resize_volume_tester.go]: Yeah that's a good point, it is updated now

File: tests/e2e/testsuites/dynamically_provisioned_resize_volume_tester.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+}
+
+// WaitForPvToResize waiting for pvc size to be resized to desired size
+func WaitForPvToResize(c clientset.Interface, ns *v1.Namespace, pvName string, desiredSize resource.Quantity, timeout time.Duration, interval time.Duration) error {
+	By(fmt.Sprintf("Waiting up to %v for pv in namespace %q to be complete", timeout, ns.Name))
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(interval) {
+		newPv, _ := c.CoreV1().PersistentVolumes().Get(pvName, metav1.GetOptions{})
+		newPvSize := newPv.Spec.Capacity["storage"]
+		if desiredSize.Equal(newPvSize) {
```
Comment (ayberk) [inline tests/e2e/testsuites/dynamically_provisioned_resize_volume_tester.go:86]: you also return an error if they're not equal, right?

Comment (AndyXiangLi) [reply tests/e2e/testsuites/dynamically_provisioned_resize_volume_tester.go:86]: Yeah, it will return error if they are not equal and timeout at line 91

Comment (ayberk) [reply tests/e2e/testsuites/dynamically_provisioned_resize_volume_tester.go:86]: Ah I see we can't exit early.

**Comment by gnufied** [comment]
@ayberk @AndyXiangLi  why are we doing this? We already had e2e via testsuites of CSI driver. I don't think we need more e2e (or if anything is missing it should be added to k/k repo).

**Comment by ayberk** [comment]
Can you point me to the e2e tests on the k/k repo? I'll make sure all new tests go there instead.

**Comment by gnufied** [comment]
here we go - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/volume_expand.go 

We should be running these tests as part of this driver's e2e.

**Pull request merged**